src/store/pageview.ts

```diff
@@ -29,37 +33,36 @@ class ViewStore<T> {
     const ts = timeStampInSecs();
 
     let keywords: string[] | null = null;
-    let topicModel: TopicModelFeature | undefined = undefined;
+    let topicDist: number[] | null = null;
     const otherFeatures: Record<string, PageFeatureValue<T>> = {};
+
     for (const pageFeature of pageFeatures) {
       if (!pageFeature.error) {
         switch (pageFeature.name) {
           case 'keywords':
-            keywords =
-              pageFeature.value instanceof Array ? pageFeature.value : null;
+            keywords = isStringArray(pageFeature.value)
+              ? pageFeature.value
+              : null;
             break;
-          case 'topicModelFeatures':
-            topicModel =
-              !(pageFeature.value instanceof Array) &&
-              pageFeature.value instanceof Object
-                ? pageFeature.value
-                : undefined;
+          case 'topicDist':
+            topicDist = isNumberArray(pageFeature.value)
+              ? pageFeature.value
+              : null;
             break;
           default:
             otherFeatures[pageFeature.name] = pageFeature.value;
         }
       }
     }
 
-    if (topicModel) {
-      return {
-        ts,
-        features: { topicModel, ...otherFeatures },
-      };
-    } else if (keywords) {
+    if (keywords || topicDist) {
       return {
         ts,
-        features: { keywords, ...otherFeatures },
+        features: {
+          ...(keywords && { keywords }),
+          ...(topicDist && { topicDist }),
+          ...otherFeatures,
```

Comment:
What happens if someone passes in `otherFeatures` ?

My guess is nothing - so why do we have this code path - it seems to add unneeded complexity.